ethash/src/compute.rs

```diff
@@ -40,9 +41,15 @@ pub struct ProofOfWork {
 	pub mix_hash: H256,
 }
 
+enum Algorithm {
+	Hashimoto,
+	Progpow(CDag),
```

Comment:
Does it make sense to `Box CDag` in the Algorithm enum? 
It is quite big (16384 bytes), thus it will make the enum `Algorithm` as big.

Reply:
Good idea. No I didn't see any regression in the benchmarks with all the changes I've done so far.